index.js

```diff
@@ -93,6 +93,27 @@ function checkTopicQueryParameter(req, res, next) {
     next();
 }
 
+
+function checkQoSParameter(req,res,next){
+
+  if(!req.body.qos){
+    req.body.qos=0
+  }
+  if( 0 > parseInt(req.body.qos, 10) || parseInt(req.body.qos, 10)  >2){
+    res.status(400).send("QoS range is not in range 0-2")
+  }
+  next();
```

Comment:
next() should not be called when qos is not in range therefore preventing other middleware/handlers from being called. (put the next() call into an else-block)